src/projections/igh_o.cpp

```diff
@@ -0,0 +1,235 @@
+
+C_NAMESPACE PJ *pj_sinu(PJ *), *pj_moll(PJ *);
+
+/* 40d 44' 11.8" [degrees] */
+/*
+static const double d4044118 = (40 + 44/60. + 11.8/3600.) * DEG_TO_RAD;
+has been replaced by this define, to eliminate portability issue:
+Initializer element not computable at load time
+*/
+#define d4044118 ((40 + 44/60. + 11.8/3600.) * DEG_TO_RAD)
+
+static const double d10  =  10 * DEG_TO_RAD;
+static const double d40  =  40 * DEG_TO_RAD;
+static const double d50  =  50 * DEG_TO_RAD;
+static const double d60  =  60 * DEG_TO_RAD;
+static const double d70  =  70 * DEG_TO_RAD;
+static const double d75  =  75 * DEG_TO_RAD;
+static const double d90  =  90 * DEG_TO_RAD;
+static const double d110 = 110 * DEG_TO_RAD;
+static const double d120 = 120 * DEG_TO_RAD;
+static const double d130 = 130 * DEG_TO_RAD;
+static const double d150 = 150 * DEG_TO_RAD;
+static const double d180 = 180 * DEG_TO_RAD;
+
+static const double EPSLN = 1.e-10; /* allow a little 'slack' on zone edge positions */
+
+namespace { // anonymous namespace
+struct pj_opaque {
+    struct PJconsts* pj[12]; \
+    double dy0;
+};
+} // anonymous namespace
+
+
+static PJ_XY igh_o_s_forward (PJ_LP lp, PJ *P) {           /* Spheroidal, forward */
+    PJ_XY xy;
+    struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
+    int z;
+
+    if (lp.phi >=  d4044118) {
+           if (lp.lam <=  -d110) z =  1;
+      else if (lp.lam >=    d50) z =  3;
+      else z = 2;
+    }
+    else if (lp.phi >=  0) {
+           if (lp.lam <=  -d110) z =  4;
+      else if (lp.lam >=    d50) z =  6;
+      else z = 5;
+    }
+    else if (lp.phi >= -d4044118) {
+           if (lp.lam <=  -d75) z =  7;
+      else if (lp.lam >=   d90) z =  9;
+      else z = 8;
+    }
+    else {
+           if (lp.lam <=  -d75) z =  10;
+      else if (lp.lam >=   d90) z =  12;
+      else z = 11;
+    }
+
+    lp.lam -= Q->pj[z-1]->lam0;
+    xy = Q->pj[z-1]->fwd(lp, Q->pj[z-1]);
+    xy.x += Q->pj[z-1]->x0;
+    xy.y += Q->pj[z-1]->y0;
+
+    return xy;
+}
+
+
+static PJ_LP igh_o_s_inverse (PJ_XY xy, PJ *P) {           /* Spheroidal, inverse */
+    PJ_LP lp = {0.0,0.0};
+    struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
+    const double y90 = Q->dy0 + sqrt(2.0); /* lt=90 corresponds to y=y0+sqrt(2) */
+
+    int z = 0;
+    if (xy.y > y90+EPSLN || xy.y < -y90+EPSLN) /* 0 */
+      z = 0;
+    else if (xy.y >=  d4044118)
+           if (xy.x <=  -d110) z =  1;
+      else if (xy.x >=    d50) z =  3;
+      else z = 2;
+    else if (xy.y >=  0)
+           if (xy.x <=  -d110) z =  4;
+      else if (xy.x >=    d50) z =  6;
+      else z = 5;
+    else if (xy.y >= -d4044118) {
+           if (xy.x <=   -d75) z =  7;
+      else if (xy.x >=    d90) z =  9;
+      else z = 8;
+    }
+    else {
+           if (xy.x <=   -d75) z =  10;
+      else if (xy.x >=    d90) z =  11;
+      else z = 12;
+    }
+
+    if (z) {
+        int ok = 0;
```

Comment:
Make this a boolean instead

Reply:
Not sure what you mean here.

Reply:
bool ok = false;

Reply:
Great - thanks @rouault.  Learning C++ as I go here.